src/openSenseLayer.js

```diff
@@ -0,0 +1,135 @@
+L.Icon.OpenSenseIcon = L.Icon.extend({
+  options: {
+    iconUrl: 'https://banner2.kisspng.com/20180409/qcw/kisspng-computer-icons-font-awesome-computer-software-user-cubes-5acb63cb589078.9265215315232787953628.jpg',
+    iconSize: [10, 10],
+    popupAnchor: [1, -34]
+  }
+});
+
+L.icon.openSenseIcon = function () {
+  return new L.Icon.OpenSenseIcon();
+};
+
+L.LayerGroup.OpenSenseLayer = L.LayerGroup.extend({
+
+  options: {
+    popupOnMouseover: true,
+    clearOutsideBounds: true
+  },
+
+  initialize: function (options) {
+    options = options || {};
+    L.Util.setOptions(this, options);
+    this._layers = {};
+  },
+
+  onAdd: function (map) {
+    this._map = map;
+    this.requestRegionData();
+  },
+
+  onRemove: function (map) {
+    this.clearLayers();
+    this._layers = {};
+  },
+
+  populatePopUp: function (e) {
+    if (e) {
+      var popup = e.target.getPopup();
+      var $ = window.jQuery;
+      var url = "https://api.opensensemap.org/boxes/" + e.target.options.boxId;
+      $.getJSON(url, function (data) {
+        var popUpContent = ""
+        if (data.name && data.grouptag) {
+          popUpContent += "<h3>" + data.name + "," + data.grouptag + "</h3>"
+        }
+        else if (data.name) {
+          popUpContent += "<h3>" + data.name + "</h3>"
+        }
+        for (var i in data.sensors) {
+          if (data.sensors[i].lastMeasurement) {
+            popUpContent += "<span><b>" + data.sensors[i].title + ": </b>" +
+              data.sensors[i].lastMeasurement.value +
+              data.sensors[i].unit + "</span><br>";
+          }
+        }
+        if (data.lastMeasurementAt) {
+          popUpContent += "<br><small>Measured at <i>" + data.lastMeasurementAt + "</i>";
+        }
+        popup.setContent(popUpContent)
+      });
+    }
+  },
+
+  requestRegionData: function () {
+    var self = this;
+
+    (function () {
+      var $ = window.jQuery;
+      var url = "https://api.opensensemap.org/boxes";
```

Comment:
Hi! This looks great, just a small question -- 

Are we requesting only one region's worth of data at a time, and how are we submitting the lat/lon coordinates? Thanks!!

Also, this PR would make a great example for #136 -- thanks!

Reply:
Thank you @jywarren . I implemented the layer in this manner owing to the [api of opensensemap](https://api.opensensemap.org/). The current implementation has used two endpoints of the api :- 

- `/boxes` - Gets data of all the boxes.
- `/boxes/:boxId` - Gets data of a particular box.

Looking at the previously implemented layers, hitting the `/boxes` endpoint seemed sufficient to create the markers and their popups, but the problem was that it didn't return the values recorded by the sensors. So `/boxes/:boxId` had to be hit to get the values. 

So whenever an user opens the `Opensense layer`, the `/boxes` endpoint is first hit which provides the coordinates and the boxID, using which the markers are created. Whenever a marker is clicked, the popup is populated at that point only by making a get request to `boxes/:boxId` that returns the values recorded by the sensors for that particular region. That is why we can see a loading text before the data is loaded.

Please let me know if it requires changes. Thanks :)